xmrstak/backend/amd/jconf.cpp

```diff
@@ -150,6 +151,13 @@ bool jconf::GetThreadConfig(size_t id, thd_cfg &cfg)
 		return false;
 	}
 
+	if(!unroll->IsUint64() || (int)unroll->GetInt64() >= 128 )
+	{
+		printer::inst()->print_msg(L0, "ERROR: unroll must be smaller than 128");
+		return false;
+	}
+	cfg.unroll = (int)unroll->GetInt64();
```

Comment:
You need to also check that unroll is a power of 2. OpenCL code will fail to compile otherwise.